File: Components/WinForms/CS/GridMainDemo/VoteModule/VotesPanel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using DevExpress.Utils.Drawing;

namespace DevExpress.VoteApp {
    public partial class VotesPanel : UserControl {
        public VotesPanel() {
            InitializeComponent();
            SetStyle(ControlStyles.AllPaintingInWmPaint | ControlStyles.ResizeRedraw | ControlStyles.OptimizedDoubleBuffer, true);
        }

        public float VotesTextSize {
            get {
                return lbVotesLeft.Font.Size;

            }
            set {
                if(VotesTextSize == value) return;
                Font fo = new Font(lbVotesLeft.Font.FontFamily, value, lbVotesLeft.Font.Style);
                lbVotesLeft.Font = fo;
                lbVotesRight.Font = fo;
            }
        }
        [Browsable(true)]
        public string VotesTextLeft {
            get { return lbVotesLeft.Text; }
            set { lbVotesLeft.Text = value ?? ""; }
        }
        public string VotesTextRight {
            get { return lbVotesRight.Text; }
            set { lbVotesRight.Text = value ?? ""; }
        }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public int VotesCountLeft { get;  set;}
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public int VotesCountRight { get; set; }
        [DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)]
        public int VotesCountTotal { get; set; }

        public Color ColorLeft { get; set; }
        public Color ColorRight { get; set; }

        void VotesPanel_Paint(object sender, PaintEventArgs e) {
            var perLeft = Helpers.GetPercent(VotesCountLeft, VotesCountTotal);
            var perRight = Helpers.GetPercent(VotesCountRight, VotesCountTotal);

            if(perLeft == 0 && perRight == 0) return;

            Rectangle bounds = ((Control)sender).ClientRectangle;
            GraphicsCache cache = new GraphicsCache(e);

            if(perLeft > 0) {
                Rectangle dem = bounds;
                dem.Width = (int)(Math.Round(dem.Width * perLeft));
                cache.FillRectangle(cache.GetSolidBrush(ColorLeft), dem);
            }
            if(perRight > 0) {
                Rectangle rep = bounds;
                rep.Width = (int)(Math.Round(rep.Width * perRight));
                rep.X = bounds.Right - rep.Width;
                cache.FillRectangle(cache.GetSolidBrush(ColorRight), rep);
            }
            Rectangle divider = bounds;
            divider.X += (int)(bounds.Width * 0.5m) - 1;
            divider.Width = 3;
            cache.FillRectangle(cache.GetSolidBrush(Color.FromArgb(200, Color.White)), divider);

        }
    }
}

File: Components/WinForms/VB/GridMainDemo/VoteModule/VotesPanel.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Drawing
Imports System.Data
Imports System.Linq
Imports System.Text
Imports System.Threading.Tasks
Imports System.Windows.Forms
Imports DevExpress.Utils.Drawing

Namespace DevExpress.VoteApp
	Partial Public Class VotesPanel
		Inherits UserControl

		Public Sub New()
			InitializeComponent()
			SetStyle(ControlStyles.AllPaintingInWmPaint Or ControlStyles.ResizeRedraw Or ControlStyles.OptimizedDoubleBuffer, True)
		End Sub

		Public Property VotesTextSize() As Single
			Get
				Return lbVotesLeft.Font.Size

			End Get
			Set(ByVal value As Single)
				If VotesTextSize = value Then
					Return
				End If
				Dim fo As New Font(lbVotesLeft.Font.FontFamily, value, lbVotesLeft.Font.Style)
				lbVotesLeft.Font = fo
				lbVotesRight.Font = fo
			End Set
		End Property
		<Browsable(True)>
		Public Property VotesTextLeft() As String
			Get
				Return lbVotesLeft.Text
			End Get
			Set(ByVal value As String)
				lbVotesLeft.Text = If(value, "")
			End Set
		End Property
		Public Property VotesTextRight() As String
			Get
				Return lbVotesRight.Text
			End Get
			Set(ByVal value As String)
				lbVotesRight.Text = If(value, "")
			End Set
		End Property
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
		Public Property VotesCountLeft() As Integer
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
		Public Property VotesCountRight() As Integer
		<DesignerSerializationVisibility(DesignerSerializationVisibility.Hidden)>
		Public Property VotesCountTotal() As Integer

		Public Property ColorLeft() As Color
		Public Property ColorRight() As Color

		Private Sub VotesPanel_Paint(ByVal sender As Object, ByVal e As PaintEventArgs) Handles MyBase.Paint
			Dim perLeft = Helpers.GetPercent(VotesCountLeft, VotesCountTotal)
			Dim perRight = Helpers.GetPercent(VotesCountRight, VotesCountTotal)

			If perLeft = 0 AndAlso perRight = 0 Then
				Return
			End If

            Dim bounds_Renamed As Rectangle = DirectCast(sender, Control).ClientRectangle
            Dim cache As New GraphicsCache(e)

			If perLeft > 0 Then
				Dim dem As Rectangle = bounds_Renamed
				dem.Width = CInt(Math.Truncate(Math.Round(dem.Width * perLeft)))
				cache.FillRectangle(cache.GetSolidBrush(ColorLeft), dem)
			End If
			If perRight > 0 Then
				Dim rep As Rectangle = bounds_Renamed
				rep.Width = CInt(Math.Truncate(Math.Round(rep.Width * perRight)))
				rep.X = bounds_Renamed.Right - rep.Width
				cache.FillRectangle(cache.GetSolidBrush(ColorRight), rep)
			End If
			Dim divider As Rectangle = bounds_Renamed
			divider.X += CInt(Math.Truncate(bounds_Renamed.Width * 0.5D)) - 1
			divider.Width = 3
			cache.FillRectangle(cache.GetSolidBrush(Color.FromArgb(200, Color.White)), divider)

		End Sub
	End Class
End Namespace
